src/lib.rs

```diff
@@ -144,6 +144,9 @@ define_if_single_runtime_enabled! {
     };
 
     pub use coll::Namespace;
+
+    #[allow(unused)]
+    pub mod change_stream;
```

Comment:
1. fixed
2. This PR only focuses on async ChangeStreams. sync version will not be possible for this internship. @saghm Correct me if I'm wrong!
3. fixed

Reply:
Yes, since we only have a limited amount of time for Utkarsh to be implementing this during his internship, I figured it would make more sense to focus on async, as sync can be added afterwards without too much work. More broadly, I tried to cut scope on the project a bit to maximize the chances that Utkarsh is able to finish by the end of the summer. Note that this PR is made to the `change-streams` branch, not to the `master` branch, which allows us to make more frequent pull requests without worrying about accidentally releasing unfinished work. I'm going to rebase master on this branch every time we commit to ensure that it doesn't have large conflicts when the work is completed.

Reply:
I'm actually not super sure about exporting ChangeStream at the top-level. For the most part, I tend to prefer keeping the top-level as slim as possible. Right now, we just have `Client`, `Database`, `Collection`, and `Cursor` in it, as those are the parts that will end up being used in virtually every application using the driver. I'd imagine that the majority of applications don't use change streams, so having it under a separate module seems reasonable to me.

Reply:
Okay, that's fine then. Sorry @utkmehta for the churn, but we can revert the re-export.

Reply:
got it